enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyServer.java

```diff
@@ -154,13 +159,25 @@ public RequestContext flushStoresAndStreamStoreFiles( String triggerName, StoreW
                 while ( files.hasNext() )
                 {
                     File file = files.next();
-                    try ( StoreChannel fileChannel = fileSystem.open( file, "r" ) )
+
+                    // Read from paged file if mapping exists. Otherwise read through file system.
+                    final Optional<PagedFile> optionalPagedFile = pageCache.getExistingMapping( file );
+                    try ( ReadableByteChannel fileChannel = optionalPagedFile.isPresent() ?
+                                                            optionalPagedFile.get().openReadableByteChannel() :
+                                                            fileSystem.open( file, "r" ) )
```

Comment:
Both `openReadableByteChannel()` and `open( file, "r" )` throw `IOException` which makes it quite messy to use `optional.map(...).orElseGet(...)`.
Do you have any suggestions on how to solve that?